### PR TITLE
Gutenboarding - Test Dynamic Previewing all Designs

### LINKED ISSUE #39248: Gutenboarding: try rendering design picker grid dynamically
Background pbAok1-cS-p2

image

- How performant it is? If it's not performant enough, what can we do to make it more performant?
  - Can we render only on hover?
  - Can we render only thumbnails that are currently visible on the page?
  - Can we render thumbnails without big images?
- Do we have what we need coming from the API?
- What kind of header/footer we can render? 
- Can we allow scrolling the whole template in thumbnail vs just show what's cropped in the thumbnail?
- Can we inject some Global styles CSS into each thumbnail?

Let's merge the functionality only behind an additional feature flag for now.

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
For summary of these `BlockPreview` draft experiments, please visit `pbAok1-lN-p2`.

### Changes proposed in this Pull Request

This branch tests dynamic previewing in the design step of Gutenboarding by using `BlockPreview` for all designs in the selector.  See #39471 for testing previews only has a hover/focus effect, #39496 for 3 at a time, and #39526 for testing global font settings.

**TLDR:** Having multiple block previews at the same time is not efficient and rendering all design options dynamically at the same time does not seem to be an option.  Loading single block previews on demand is a possibility but doing so independently from the editor will require us to include 3rd party block support, theme styles and settings, etc.

**More Info**
Currently the previews render broken layouts due to lack of 3rd party block support and missing theme styles and settings.  However, they are running and rendering some images so they are set up to give us a good idea about what could be possible in future iterations of Gutenboarding designs.

Some observations thus far:
1.  Running Block Preview for 18 templates at once is not performant and takes a considerable amount of time to load.
2.  Running Block Preview for 2 templates 9 times each does not seem to make a notable improvement in load times.  Thus having templates share images to match the site vertical will not make this any better.  
3.  Parsing blocks out of the content strings in all templates seems to take about 1-2 seconds on its own.  This can be seen by leaving in the parsing functions while commenting out the `DynamicPreview` component in favor of the old cards.  In gutenboarding, go 'back' to the intent page and click 'continue' again.  You should notice a delay before the design page is loaded.  (This delay can be noticed in SPTs as well by opening them through the sidebar and verifying the modal).

Im thinking we may want to limit Block Previews to one at a time ( and best case would be to load one previewer to then use for all previews as in starter page templates ).   I think having scrollable static images per vertical / design will serve us better for a step like design/page selectors.

### Testing instructions

* Run this Branch in Calypso.
* Go to the design page to see the 18 (kinda broken) dynamic previews.
* If desired to explore further, adjust Comments in `design-selector/index.tsx` to try rendering different groups of templates or repeat templates, as well as rendering the old Cards instead to see the impact of Parsing blocks alone.

Fixes #39248

### CODE REVIEW

**Comment by simison** [comment]
> Im thinking we may want to limit Block Previews to one at a time ( two is probably possible, but having multiple previews is not advised ). I think having scrollable static images per vertical / design will serve us better for a step like design/page selectors.

Would rendering up to three previews at once be performant?

image

**Comment by Addison-Stavlo** [comment]
> Would rendering up to three previews at once be performant?

I don't think so really.  I have a draft to mimic that behavior - https://calypso.live/?branch=try/gutenboarding-preview-3-aat - (#39496).  Everything seems fairly bogged down and laggy from the system end and this will only get worse on lower end devices.
